File: 1249-minimum-remove-to-make-valid-parentheses/1249-minimum-remove-to-make-valid-parentheses.py
class Solution:
    def minRemoveToMakeValid(self, s: str) -> str:
        string = list(s)
        stack = list()

        for i in range(len(string)):
            if string[i] == '(':
                stack.append(i)
            elif string[i] == ')':
                if stack:
                    stack.pop()
                else:
                    string[i] = ''

        while stack:
            string[stack.pop()] = ''

        return ''.join(string)

File: 1249-minimum-remove-to-make-valid-parentheses/1249-minimum-remove-to-make-valid-parentheses.java
class Solution {
    public String minRemoveToMakeValid(String s) {
        char[] str = s.toCharArray();
        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < str.length; i++) {
            if (str[i] == '(') {
                stack.push(i);
            } else if (str[i] == ')') {
                if (stack.isEmpty()) {
                    str[i] = '*';
                } else {
                    stack.pop();
                }
            }
        }

        while (!stack.isEmpty()) {
            str[stack.pop()] = '*';
        }

        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < str.length; i++) {
            if (str[i] != '*') {
                sb.append(str[i]);
            }
        }

        return sb.toString();
    }
}